src/shared/validation/use-imperative-validate.js

```diff
@@ -0,0 +1,52 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { useApiAttributeParams } from '../use-api-attribute-params.js'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../use-context-selection/index.js'
+import buildValidationResult from './build-validation-result.js'
+import {
+    getValidationMetaDataQueryKey,
+    getValidationQueryKey,
+} from './query-key-factory.js'
+
+/**
+ * Returns validation rule violations grouped by importance
+ * @params {Object} options
+ * @params {Boolean} options.enabled - Defaults to `true`
+ **/
+export default function useImperativeValidate() {
+    const client = useQueryClient()
+    const [dataSetId] = useDataSetId()
+    const [periodId] = usePeriodId()
+    const [orgUnitId] = useOrgUnitId()
+    const {
+        attributeCombo: categoryComboId,
+        attributeOptions: categoryOptionIds,
+    } = useApiAttributeParams()
+
+    return () => {
+        const validationQueryKey = getValidationQueryKey({
+            dataSetId,
+            periodId,
+            orgUnitId,
+            categoryComboId,
+            categoryOptionIds,
+        })
+        const validationMetaDataQueryKey =
+            getValidationMetaDataQueryKey(dataSetId)
+
+        const vQ = client.fetchQuery(validationQueryKey)
+        const vMDQ = client.fetchQuery(validationMetaDataQueryKey)
+
+        return Promise.all([vQ, vMDQ]).then(
```

Comment:
Errors are handled by the `<CompleteButton />` so we can show an appropriate message to the user and set the local loading state to false in case of an error